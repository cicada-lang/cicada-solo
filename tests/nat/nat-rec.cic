add(x: Nat, y: Nat): Nat {
  nat_rec(y, x, (prev, almost) { add1(almost) })
}

mul(x: Nat, y: Nat): Nat {
  nat_rec(y, 0, (prev, almost) { add(almost, x) })
}

pow(x: Nat, y: Nat): Nat {
  nat_rec(y, 1, (prev, almost) { mul(almost, x) })
}

eight =
  four = add(2, 2)
  add(four, four)

the(Equal(Nat, eight, pow(2, 3)), same)
