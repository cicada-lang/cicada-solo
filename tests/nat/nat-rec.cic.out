same: Equal(Nat, 8, 8)
